presto-main/src/main/java/io/prestosql/operator/CyclingGroupByHash.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.spi.Page;
+import io.prestosql.spi.PageBuilder;
+import io.prestosql.spi.block.BlockBuilder;
+import io.prestosql.spi.type.Type;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.List;
+
+import static io.prestosql.spi.type.BigintType.BIGINT;
+
+/**
+ * GroupByHash that provides a round robin group ID assignment.
+ */
+public class CyclingGroupByHash
+        implements GroupByHash
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(CyclingGroupByHash.class).instanceSize();
+
+    private final int totalGroupCount;
+    private int maxGroupId;
+    private int currentGroupId;
+
+    public CyclingGroupByHash(int totalGroupCount)
+    {
+        this.totalGroupCount = totalGroupCount;
+    }
+
+    @Override
+    public long getEstimatedSize()
+    {
+        return INSTANCE_SIZE;
+    }
+
+    @Override
+    public long getHashCollisions()
+    {
+        return 0;
+    }
+
+    @Override
+    public double getExpectedHashCollisions()
+    {
+        return 0;
+    }
+
+    @Override
+    public List<Type> getTypes()
+    {
+        return ImmutableList.of();
+    }
+
+    @Override
+    public int getGroupCount()
+    {
+        return maxGroupId;
+    }
+
+    @Override
+    public void appendValuesTo(int groupId, PageBuilder pageBuilder, int outputChannelOffset)
+    {
+        throw new UnsupportedOperationException("CyclingGroupByHash does not support appendValuesTo");
+    }
+
+    @Override
+    public Work<?> addPage(Page page)
+    {
+        // Create a dump work whose result is never used.
+        return new CompletedWork<>(0);
```

Comment:
For benchmarking I think you can use a JMH `Blackhole` for the sink

Reply:
@dain, can you clarify this? doesn't returning the value from a benchmarked method effectively act as a blackhole?

Reply:
You have a comment `// Create a dump work whose result is never used.`,  If you want to dump a result that is not used in a benchmark, `Blackhole` can be used.  Also this constructs an object which has some cost... Anyway, it was just a suggestion based on the comment; there is no need to change anything

Reply:
> // Create a dump work whose result is never used.

Oh, this comes from NoChannelGroupByHash (which I copied). I never call this, so I can just throw an UnsupportedOperationException if you think that would be better